### ISSUE TITLE
[SKIA] [WebAssembly] Using ComboBox the ItemsPanelRoot.Children is not updated after removing and adding new items

### ISSUE THREAD

**Issue by CesarRabelo** [unoplatform/uno#8204]
### Current behavior

I have a ComboBox control with several entries. Those entries are localized and set via XAML.
When I remove one of the entries, the SelectedIndex properties does not reflect the correct index anymore, but either returns -1 or the original index when the collection was created.
Also setting the selection via index is messed up afterwards.

And if I have 6 items for example
![image](https://user-images.githubusercontent.com/790876/156056042-ecb07937-2774-41a6-bf9e-16a355d1b837.png)

If I remove item 2 the popup remove last item.
![image](https://user-images.githubusercontent.com/790876/156056126-9adebb39-768e-40e1-8064-b068ab6ed9fb.png)

The samples used step by step
![image](https://user-images.githubusercontent.com/790876/156056330-2c694eb9-ac2a-4cae-9983-1fb5572dc251.png)

### Expected behavior

The index reflects the correct position of the item within the Items collection.
The popup presents the correct list of items after remove or add items.
![image](https://user-images.githubusercontent.com/790876/156058934-3691c363-3085-4082-a3a5-0127a4502a39.png)

### How to reproduce it (as minimally and precisely as possible)

``` 
<StackPanel Orientation="Vertical">
        <TextBlock Text="{Binding ElementName=combo, Path=SelectedIndex}"/>
        <ComboBox x:Name="combo">
            <ComboBoxItem>Item 0</ComboBoxItem>
            <ComboBoxItem>Item 1</ComboBoxItem>
            <ComboBoxItem>Item 2</ComboBoxItem>
            <ComboBoxItem>Item 3</ComboBoxItem>
            <ComboBoxItem>Item 4</ComboBoxItem>
            <ComboBoxItem>Item 5</ComboBoxItem>
        </ComboBox>
    </StackPanel>
```

And simply call in the page's code behind the following line:
combo.Items.RemoveAt(2);

Check  this properties

```
combo.ItemsPanelRoot.Children.Count

ComboBoxItem cbi = combo.ContainerFromIndex(combo.SelectedIndex) as ComboBoxItem;
cbi.Content;
```

### Workaround

Clear the collection (combo.Items.Clear()) and add items again, but that's not really fancy.😉

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

4.0.11

### Affected platforms

Skia (WPF), Skia (GTK on Linux/macOS/Windows)

### IDE

Visual Studio 2022

### IDE version

17.0.5

**Comment by carldebilly** [comment]
@CesarRabelo Should be fixes with https://github.com/unoplatform/uno/pull/8227

**Comment by CesarRabelo** [comment]
Closed because issue was solved in #8227
